### PR TITLE
show private heatmap on profile to self & admins

### PR DESCRIPTION
**Pull request by noerw**
followup to #14080, i thought i changed that back then..

this could go with skip-changelog into 1.14, but is not critical i guess

### CODE REVIEW

**Review by wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf — APPROVED** [review]
works great 🚀 
show's heatmap even when it's empty (which it should).

admin's view of a user with disabled public activity (essentially the user themselves sees the same):
image

non-empty user's view heatmap:
image

other users, of course, see only the message telling that the user has public activity disabled.

EDIT: added user's view with non-empty activity.

**Comment by 6543** [comment]
@noerw look's like test's fail related

**Review by 6543 — APPROVED** [review]

File: templates/user/heatmap.tmpl [hunk 1]
```diff
@@ -1,8 +1,6 @@
-{{if .HeatmapData}}
-	<div id="user-heatmap" data-heatmap-data="{{Json .HeatmapData}}">
```
Comment (6543) [inline templates/user/heatmap.tmpl]: can you revert this file back to master

or what was the reason to do this?

**Review by 6543 — CHANGES_REQUESTED** [review]

File: templates/user/heatmap.tmpl (hunk 1 above)
Comment (noerw) [reply templates/user/heatmap.tmpl]: It's not directly related, I just stumbled over this. From the commit message:
> always show heatmp, even if it contains no data 
> 
> rendering is wanted for freshly created users.
> rendering might be unwanted, if doer has no permission to view anyway

If you think this behaviour isn't better we can revert that

Comment (6543) [reply templates/user/heatmap.tmpl]: I dont like the new behavior

**Comment by 6543** [comment]
cc @noerw 
```
--- FAIL: TestPrivateActivityYesHeatmapVisibleForUserItselfAtDashboard (0.20s)                                                                                                                                                                                                 
    testlogger.go:78: 2021/03/04 21:34:36 ...uters/routes/base.go:35:1() [I] Started GET /user/settings/applications for                                                                                                                                                       
    testlogger.go:78: 2021/03/04 21:34:36 ...uters/routes/base.go:44:1() [I] Completed GET /user/settings/applications 200 OK in 1.907951ms                                                                                                                                    
    testlogger.go:78: 2021/03/04 21:34:36 ...uters/routes/base.go:35:1() [I] Started POST /user/settings/applications for                                                                                                                                                      
    testlogger.go:78: 2021/03/04 21:34:36 ...uters/routes/base.go:44:1() [I] Completed POST /user/settings/applications 302 Found in 10.237917ms                                                                                                                               
    testlogger.go:78: 2021/03/04 21:34:36 ...uters/routes/base.go:35:1() [I] Started GET /user/settings/applications for                                                                                                                                                       
    testlogger.go:78: 2021/03/04 21:34:36 ...uters/routes/base.go:44:1() [I] Completed GET /user/settings/applications 200 OK in 1.596781ms                                                                                                                                    
    testlogger.go:78: 2021/03/04 21:34:36 ...uters/routes/base.go:35:1() [I] Started POST /api/v1/repos/user2/repo1/issues?state=all&token=17e09f6fc3fb4cff9f74333f7680d077148f3e61 for                                                                                        
    testlogger.go:78: 2021/03/04 21:34:36 ...vices/mailer/mail.go:247:composeIssueCommentMessages() [E] ExecuteTemplate [issue/default/body]: html/template: "issue/default" is an incomplete template                                                                         
    testlogger.go:78: 2021/03/04 21:34:36 ...uters/routes/base.go:44:1() [I] Completed POST /api/v1/repos/user2/repo1/issues?state=all&token=17e09f6fc3fb4cff9f74333f7680d077148f3e61 201 Created in 14.841541ms                                                               
    testlogger.go:78: 2021/03/04 21:34:36 ...uters/routes/base.go:35:1() [I] Started GET /user/settings for                                                                                                                                                                    
    testlogger.go:78: 2021/03/04 21:34:36 ...uters/routes/base.go:44:1() [I] Completed GET /user/settings 200 OK in 1.607331ms                                                                                                                                                 
    testlogger.go:78: 2021/03/04 21:34:36 ...uters/routes/base.go:35:1() [I] Started POST /user/settings for                                                                                                                                                                   
    testlogger.go:78: 2021/03/04 21:34:36 ...uters/routes/base.go:44:1() [I] Completed POST /user/settings 302 Found in 959.081µs                                                                                                                                              
    testlogger.go:78: 2021/03/04 21:34:36 ...uters/routes/base.go:35:1() [I] Started GET / for                                                                                                                                                                                 
    testlogger.go:78: 2021/03/04 21:34:36 ...uters/routes/base.go:44:1() [I] Completed GET / 200 OK in 3.262753ms                                                                                                                                                              
    privateactivity_test.go:303:                                                                                                                                                                                                                                               
                Error Trace:    privateactivity_test.go:303                                                                                                                                                                                                                    
                Error:          Should be true                                                                                                                                                                                                                                 
                Test:           TestPrivateActivityYesHeatmapVisibleForUserItselfAtDashboard                                                                                                                                                                                   
                Messages:       user should have visible heatmap
```

**Review by 6543 — APPROVED** [review]
:rocket:

**Review by zeripath — APPROVED** [review]

**Pull request merged**